### DISCUSSION TITLE
Tun2Socks in Windows Then to Android Phone via Ethernet Cable

### DISCUSSION THREAD

**Discussion by matevzmlakar** [xjasonlyu/tun2socks#257] [General]
Hey guys,

So very briefly, I am SSTAP Beta user. My main concern why I am here is that I use SSTAP Beta with SOCKS5 to make na connection on Network Layer in my windows and then I connect my phone with cable via Ethernet. 

I am here becausue SSTAP Beta does not support IPV6 connections. 

I was speaking witha Jason, he told me this can be resolvable with tun2sock and proxy that support ipv4 and ipv6 connections. But I assume I will also need something to create connections on Network Layer? What can I use in this case?

If there is anyone who can give me a help with that will be forever grateful. I am also willing to pay one hour of your time, and you can show me how the things work via Teamviewer. This would be something that I can dream of and will help the most. Because I do have such a hussle with this problem lately and I know for someone who is this kind of computer savy is something easy to do. 

Thank you so much!

Matt

**Comment by xjasonlyu** [comment]
Try these steps to setup tun2socks in Windows:

- Download wintun.dll and put in system PATH or the same folder with tun2socks binary.
- Start tun2socks via cmd: `tun2socks -device wintun -proxy socks5://<your_socks_server_ip>:<port>`
- Configure routing tables:
  - IPv4: `netsh routing ip add persistentroute <network_address> <netmask> <interface> <local_address>`
  - IPv6: `netsh interface ipv6 add route <network_address/prefixlength> <interface>`
 
To setup your proxy server, try something like [dante](https://www.inet.no/dante/).

**Reply by xjasonlyu** [reply]
See https://github.com/xjasonlyu/tun2socks/wiki/Examples#windows
